gcalbot/gcalbot/schedulescheduler/send.go

```diff
@@ -0,0 +1,158 @@
+	if err != nil {
+		s.Errorf("error getting daily schedule subscriptions to sync: %s", err)
+	}
+	subscriptions = append(subscriptions, tomorrowSubscriptions...)
+
+	for _, subscription := range subscriptions {
+		select {
+		case <-shutdownCh:
+			return
+		default:
+		}
+
+		s.SendDailyScheduleMessage(sendMinute, subscription)
+	}
+
+	s.stats.Value("sendDailySchedulesForMinute - duration - seconds", time.Since(sendMinute).Seconds())
+}
+
+func (s *ScheduleScheduler) SendDailyScheduleMessage(sendMinute time.Time, subscription *gcalbot.AggregatedDailyScheduleSubscription) {
+	userSendMinute := sendMinute.In(subscription.Timezone)
+	dayStart := time.Date(sendMinute.Year(), sendMinute.Month(), sendMinute.Day(), 0, 0, 0, 0, subscription.Timezone)
+	notificationDuration := sendMinute.Sub(dayStart).Round(time.Minute)
+	if notificationDuration.Minutes() != subscription.NotificationDuration.Minutes() {
+		return
+	}
+
+	switch subscription.DaysToSend {
+	case gcalbot.DaysToSendEveryday:
+	case gcalbot.DaysToSendMonToFri:
+		switch userSendMinute.Weekday() {
+		case time.Saturday, time.Sunday:
+			return
+		}
+	case gcalbot.DaysToSendSatToThu:
+		switch userSendMinute.Weekday() {
+		case time.Friday:
+			return
+		}
+	}
+
+	srv, err := gcalbot.GetCalendarService(&subscription.Account, s.oauth)
+	if err != nil {
+		s.Errorf(err.Error())
+		return
+	}
+
+	minTime := dayStart
+	switch subscription.ScheduleToSend {
+	case gcalbot.ScheduleToSendToday:
+	case gcalbot.ScheduleToSendTomorrow:
+		minTime = time.Date(userSendMinute.Year(), userSendMinute.Month(), userSendMinute.Day()+1, 0, 0, 0, 0, subscription.Timezone)
+	}
+	maxTime := minTime.Add(24 * time.Hour)
+
+	format24HourTime, err := gcalbot.GetUserFormat24HourTime(srv)
+	if err != nil {
+		s.Errorf("unable to get user 24 hour time setting: %s", err)
+		return
+	}
+
+	calendarSummaries := make([]string, len(subscription.CalendarIDs))
+	var events []*calendar.Event
+	for index, calendarID := range subscription.CalendarIDs {
+		cal, err := srv.Calendars.Get(calendarID).Fields("summary").Do()
+		if err != nil {
+			s.Errorf("error getting calendar summary from API: %s", err)
+			calendarSummaries[index] = calendarID // use the cal id if there is an error
+		} else {
+			calendarSummaries[index] = cal.Summary
+		}
+		err = srv.Events.
+			List(calendarID).
+			TimeMin(minTime.Format(time.RFC3339)).
+			TimeMax(maxTime.Format(time.RFC3339)).
+			SingleEvents(true).
+			OrderBy("startTime").
+			Pages(context.Background(), func(page *calendar.Events) error {
+				events = append(events, page.Items...)
+				return nil
+			})
+		if err != nil {
+			s.Errorf("error getting events from API: %s", err)
+			continue
+		}
+	}
+
+	// TODO(marcel): better calendar listing
+	calendarList := strings.Join(calendarSummaries, ", ")
+	if len(events) == 0 {
+		s.ChatEcho(subscription.KeybaseConvID, "You have no events for the calendar(s) %s in account '%s' for %s.",
```

Comment:
did you not push something? i thought we settled with a different format over chat

Reply:
Yeah I forgot to push 😅